### PR TITLE
Cover Map Locate with Unit Tests

### LINKED ISSUE #8370: Implement test for 'locate' method of Map

### PR DESCRIPTION
**Pull request by stephenspol**
Test Cover `locate` method in `Map.js`. 

I saw there already was a `describe` for `#Geolocation`. Let me know if you would  like me to put these tests within the `describe` block. I am also not testing private methods related to Geolocation but I can if that is desired.

PR closes #8370.

### CODE REVIEW

File: spec/suites/map/MapSpec.js [hunk 1]
```diff
@@ -2032,4 +2032,171 @@ describe("Map", function () {
...
+			map.locate({setView: true});
+
+			expect(errorSpy.called).to.be.ok();
+			expect(foundSpy.called).to.not.be.ok();
+		});
+
+		it("sets map view to geolocation coords", function () {
+			Object.defineProperty(window, 'navigator', {
+				value: geolocationStub
+			});
```
Comment (stephenspol) [inline spec/suites/map/MapSpec.js:2112]: Let me know if there is a better way to stub geolocation.

File: spec/suites/map/MapSpec.js [hunk 2]
```diff
@@ -2032,4 +2032,171 @@ describe("Map", function () {
...
+
+			var expectedBounds;
+
+			map.on("locationfound", function (data) {
+				expect(data.latlng).to.eql(L.latLng(50, 50));
```
Comment (Falke-Design) [inline spec/suites/map/MapSpec.js:2117]:
```suggestion
				expect(data.latlng).to.be.nearLatLng(expectedLatLngs);
```

File: spec/suites/map/MapSpec.js [hunk 3]
```diff
@@ -2032,4 +2032,171 @@ describe("Map", function () {
...
+
+			map.locate({setView: true});
+
+			expect(errorSpy.called).to.not.be.ok();
+			expect(foundSpy.called).to.be.ok();
+
+			expect(getCurrentPosSpy.called).to.be.ok();
+			expect(watchPosSpy.called).to.not.be.ok();
+
+			expect(map.getCenter().distanceTo([50, 50])).to.be(0);
```
Comment (Falke-Design) [inline spec/suites/map/MapSpec.js:2135]: Please use `nearLatLng` and expectedLatLngs

File: spec/suites/map/MapSpec.js [hunk 4]
```diff
@@ -2032,4 +2032,171 @@ describe("Map", function () {
...
+			expect(errorSpy.called).to.be.ok();
+			expect(foundSpy.called).to.not.be.ok();
+		});
+
+		it("sets map view to geolocation coords", function () {
+			Object.defineProperty(window, 'navigator', {
+				value: geolocationStub
+			});
+
+			var expectedBounds;
```
Comment (Falke-Design) [inline spec/suites/map/MapSpec.js:2114]:
```suggestion
			var expectedBounds;
			var expectedLatLngs = [50, 50];
```

File: spec/suites/map/MapSpec.js [hunk 5]
```diff
@@ -2032,4 +2032,171 @@ describe("Map", function () {
...
+
+		it("sets map view to geolocation coords and returns location watch ID when watch is true", function () {
+			Object.defineProperty(window, 'navigator', {
+				value: geolocationStub
+			});
+
+			var expectedBounds;
+
+			map.on("locationfound", function (data) {
+				expect(data.latlng).to.eql(L.latLng(25, 25));
```
Comment (Falke-Design) [inline spec/suites/map/MapSpec.js:2150]: Please use nearLatLng and expectedLatLngs

File: spec/suites/map/MapSpec.js [hunk 6]
```diff
@@ -2032,4 +2032,171 @@ describe("Map", function () {
...
+
+			map.locate({setView: true, watch: true});
+
+			expect(errorSpy.called).to.not.be.ok();
+			expect(foundSpy.called).to.be.ok();
+
+			expect(getCurrentPosSpy.called).to.not.be.ok();
+			expect(watchPosSpy.called).to.be.ok();
+
+			expect(map.getCenter().distanceTo([25, 25])).to.be(0);
```
Comment (Falke-Design) [inline spec/suites/map/MapSpec.js:2168]: Please use nearLatLng and expectedLatLngs

File: spec/suites/map/MapSpec.js [hunk 7]
```diff
@@ -2032,4 +2032,171 @@ describe("Map", function () {
...
+			expect(map._locationWatchId).to.eql(25);
+		});
+
+		it("does not set map view by default", function () {
+			Object.defineProperty(window, 'navigator', {
+				value: geolocationStub
+			});
+
+			map.on("locationfound", function (data) {
+				expect(data.latlng).to.eql(L.latLng(50, 50));
```
Comment (Falke-Design) [inline spec/suites/map/MapSpec.js:2183]: Please use nearLatLng

**Review by Falke-Design — CHANGES_REQUESTED** [review]
Looks very nice, only a few small changes

**Review by Falke-Design — APPROVED** [review]
Great, thank you!

**Pull request merged**